### PR TITLE
Introduce `chainerx.ndarray._is_chained`

### PR DESCRIPTION
**Pull request by hvy**
In Chainer, we can check the existence of the creator of a Variable to find out if it's a leaf node in the computational graph. There is no equivalent feature in ChainerX. This PR introduces one.

It will be needed in https://github.com/chainer/chainer/pull/7507 since warnings/errors will be raised depending on this state.

### CODE REVIEW

File: chainerx_cc/chainerx/array.cc [hunk 1]
```diff
@@ -464,6 +464,14 @@ bool Array::IsGradRequired(const nonstd::optional<BackpropId>& backprop_id) cons
     return body_->IsGradRequired(actual_backprop_id);
 }
 
+bool Array::IsUnchained(const nonstd::optional<BackpropId>& backprop_id) const {
+    BackpropId actual_backprop_id = internal::GetArrayBackpropId(*this, backprop_id);
+    if (!body_->HasArrayNode(actual_backprop_id)) {
+        throw ChainerxError{"Array is constant with respect to the computation for backprop ID: '", actual_backprop_id, "'."};
+    }
```
Comment (hvy) [inline chainerx_cc/chainerx/array.cc]: Maybe simpler to return false instead of throwing. Open for discussion.

**Comment by niboshi** [comment]
Some thoughts:

* Currently this feature is only needed for implementing `chainer.Link.to_device()`, and we don't have any other use cases in mind, do we? So i think we can make it a private feature for now (i.e. prefix with `_`).

* There seems to be a room for discussion about the name `is_unchained`. First of all we should avoid negative prefix like `un` in a method that queries a logical value, as a general naming practice. The term "chained" might be ambiguous as well. Some random candidates include: `is_chained`, `is_connected`, `has_input_nodes`, `has_upstream_nodes`, `is_input`, `is_input_leaf`

**Comment by hvy** [comment]
Made the method private as suggested and changed the name to one of the proposals above `_is_chained`. I think this name is fine as one may relate to "unchain" and its additionally private.

**Comment by hvy** [comment]
PTAL.

**Comment by hvy** [comment]
Added a comment. PTAL.

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
LGTM

**Pull request merged**